src/items/generics.md

```diff
@@ -83,6 +83,98 @@ where
 }
 ```
 
+## Formulae and Free Variables
+
+The syntatic locations a *variable* is allowed is a *formulae*. Formulae for
+type and lifetime variables are:
+
+* [Types]
+* Type constructors (for example, `Option` in `enum Option<T> { ... }`)
+* Lifetimes (only lifetime variables)
+* [Items]
+* Item constructors (for example, `Into` in `trait Into<T>`)
+* [Associated Items]
+* [Trait Bounds]
+* [Expressions] \(through path expressions with generics)
+
+The free variables of a formulae are those not referenced (*bound*) by a generic
+binder in the formulae.
+
+These free variables may be substituted for other types or lifetimes.
+A formulae is *concrete*, or *closed*, if there are no free variables in it.
+See [Wikipedia][wikipedia free variables] for more.
+
+Examples of generic binders for type and lifetime variables are:
+
+* `for<'a>` in `for<'a> fn(&'a u8)`
+* `<T>` in `fn identity<T>(x: T) -> T { x }` for item constructors
+* [Traits] and [implementations] implicitly bind `Self`
+
+All type and lifetimes are unbound at the boundaries of items.
+
+Formulae nest. For example, in a function item, the entire function item is a
+formulae but so are the types of the parameters of the function. A type or
+lifetime variable may be free in one formulae but bound in a containing
+formulae. For example, in the function prototype `for<'a> fn foo(a: &'a i32>`,
```

Comment:
```suggestion
formulae. For example, in the function prototype `for<'a> fn foo(a: &'a i32)`,
```
"prototype"?

Reply:
Name for everything of a function except the block expression. The part you can put in a trait. Do you have a better name?

Reply:
I'd call this "function signature" or "function declaration".